### DISCUSSION TITLE
Reverse Geocoding jobs keep failing.

### DISCUSSION THREAD

**Discussion by dunc1ca** [Freika/dawarich#399] [Q&A]
I'm not sure if I'm doing something wrong but I have approximately 500 thousand points to reverse geocode.  After 3 weeks, only about 20 thousand have been successful. The rest will fail and go through a retry cycle.  I have cleared the queue and selected continue reverse geocoding since upgrading versions, but it still continues to fail most geocode jobs.

**Comment by dunc1ca** [comment]
| Next Retry | Retry Count | Queue | Job | Arguments | Error
-- | -- | -- | -- | -- | -- | --
  | just now | 0 | reverse_geocoding | ReverseGeocodingJob | "Point", 618367 | Geocoder::LookupTimeout: Geocoder::LookupTimeout
  | right now | 0 | reverse_geocoding | ReverseGeocodingJob | "Point", 618397 | Errno::ECONNREFUSED: Errno::ECONNREFUSED

**Comment by shaman007** [comment]
Seems like they're throttling and banning IP's. There are several Dawarich instanses seems ti be started and they see some IP's using the API.

I'd checked, to set up OSM locally about 128 GB RAM is needed and severe effort to be applied.

UPD:

the thing to handle this stuff woudl be about 100 Eur/month, better 150.

**Reply by shaman007** [reply]
Would be really cool to get in touch with @jonaskmt or @komoot if they have some API pricing for enthusiasts. I am sure they don't need my 100-200-300 eur badly to allow to geocode stuff, still I would rather pay a bit to have API access.

**Reply by shaman007** [reply]
Welp, downloading photon now. See https://github.com/komoot/photon/discussions/848

**Comment by Freika** [comment]
Added a section on that to the FAQ: https://dawarich.app/docs/FAQ#why-i-have-so-many-failed-jobs-in-sidekiq

Also, we have an instruction on how to set up your own Photon instance to be independent of Nominatim's API limits: https://dawarich.app/docs/tutorials/reverse-geocoding#setting-up-your-own-reverse-geocoding-service

**Comment by Freika** [comment]
@shaman007 Photon doesn't need 128GB of RAM, since it provides only geocoding/reverse geocoding api, it's heavy on disk space but not on RAM

**Reply by shaman007** [reply]
Yeah, I'd already tried my own. With only me as a user it's totaly managable.

**Reply by Freika** [reply]
Yeah, I see you got an answer in Photon's repo :)

**Comment by dunc1ca** [comment]
I seem to have fixed the reverse geocoding fails by changing BACKGROUND_PROCESSING_CONCURRENCY: from 10 to 1.  It seems like komoot didn't like the request rate and just rejected me.  It is now geocoding about 4000 points per hour.
